src/main/java/org/candlepin/policy/js/compliance/hash/HashableStringGenerators.java

```diff
@@ -212,13 +222,21 @@ public String generate(Consumer target) {
 
             String generated = target.getUuid();
 
-            generated += generateFromCollection(target.getFacts() != null ?
-                target.getFacts().entrySet() : null, STRING_ENTRY);
+            generated += generateFromCollection(factsOf(target), STRING_ENTRY);
             generated += generateFromCollection(target.getInstalledProducts(), INSTALLED_PRODUCT);
             generated += generateFromCollection(target.getEntitlements(), ENTITLEMENT);
             return generated;
         }
 
+        private Collection<Entry<String, String>> factsOf(Consumer target) {
```

Comment:
This method could be added directly onto the Enum

Reply:
Done.